src/Microsoft.Identity.Web/WebAppExtensions/MicrosoftIdentityWebAppAuthenticationBuilder.cs

```diff
@@ -179,8 +179,31 @@ internal static void WebAppCallsWebApiImplementation(
 
                                if (clientInfoFromServer != null && clientInfoFromServer.UniqueTenantIdentifier != null && clientInfoFromServer.UniqueObjectIdentifier != null)
```

Comment:
The rejection logic should work for if either claim is present, but right now it requires both. This needs to be restructured.
```cs
if (clientInfoFromServer is not null)
{
    bool hasServerUti = ...;
    bool hasServerUoi = ...;
    ...
}
```

Reply:
@bgavrilMS should this be tackled in this PR? I believe this seems to be another enhancement related but out of scope.

Thoughts?

Reply:
I think this is in scope since if only one of the internal claims is present then the problem documented in the issue for which this PR was created will remain.